cmd/terramate/e2etests/cmd/test/test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// test is a test command that implements behaviors that are
+// useful when testing terramate run features in a way that reduces
+// dependencies on the environment to run the tests.
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		log.Fatal("test requires at least one subcommand argument")
+	}
+
+	switch os.Args[1] {
+	case "hang":
+		hang()
+	default:
+		log.Fatalf("unknown command %s", os.Args[1])
+	}
+}
+
+// hang will hang the process forever, ignoring any signals.
+// It is useful to validate forced kill behavior.
+// It will print "ready" when it starts to receive the signals.
+// It will print the name of the received signals, which may also be useful in testing.
+func hang() {
+	signals := make(chan os.Signal, 10)
```

Comment:
I don't think so, it is tricky. The only recommendation done from Go (and linters) is that the channel needs to be at least buffered, but it is not a game you can win, even if the buffer is quite big there is always a chance that if you get a lot of signals before reading you may lose something, so you are always just improving your odds. At least on the tests we rarely send that much signals, so it seemed like a reasonably safe way to avoid problems. WDYT ?

Reply:
yeah keep that way.